packages/utils/src/types.ts

```diff
@@ -246,6 +248,26 @@ export interface TemplatesType<T = any, S extends StrictRJSFSchema = RJSFSchema,
   };
 }
 
+/** The set of UiSchema options that can be set globally and used as fallbacks at an individual template, field or
+ * widget level when no field-level value of the option is provided.
+ */
+export type GlobalUISchemaOptions = {
+  /** Flag, if set to `false`, will mark array fields as NOT being able to be added to (defaults to true) */
+  addable?: boolean;
+  /** Flag, if set to `true`, will mark array fields as being able to be copied (defaults to false) */
+  copyable?: boolean;
+  /** Flag, if set to `false`, will mark array fields as NOT being able to be ordered (defaults to true) */
+  orderable?: boolean;
+  /** Flag, if set to `false`, will mark array fields as NOT being able to be removed (defaults to true) */
+  removable?: boolean;
+  /** Field labels are rendered by default. Labels may be omitted by setting the `label` option to `false` */
+  label?: boolean;
+  /** When using `additionalProperties`, key collision is prevented by appending a unique integer to the duplicate key.
+   * This option allows you to change the separator between the original key name and the integer. Default is "-"
+   */
+  duplicateKeySuffixSeparator?: string;
```

Comment:
Trying to improve clarity by focusing on the outcome of the flag, take it or leave it
```suggestion
  /** Flag, if set to `false`, new items cannot be added to array fields, unless overridden (defaults to true) */
  addable?: boolean;
  /** Flag, if set to `true`, array items can be copied (defaults to false) */
  copyable?: boolean;
  /** Flag, if set to `false`, array items cannot be ordered (defaults to true) */
  orderable?: boolean;
  /** Flag, if set to `false`, array items will not be removable (defaults to true) */
  removable?: boolean;
  /** Field labels are rendered by default. Labels may be omitted by setting the `label` option to `false` */
  label?: boolean;
  /** When using `additionalProperties`, key collision is prevented by appending a unique integer to the duplicate key.
   * This option allows you to change the separator between the original key name and the integer. Default is "-"
   */
  duplicateKeySuffixSeparator?: string;
```